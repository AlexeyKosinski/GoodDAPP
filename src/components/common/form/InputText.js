// @flow
import React from 'react'
import { StyleSheet, TextInput } from 'react-native'
import { withStyles } from '../../../lib/styles'

const InputText = (props: any) => {
  return <TextInput {...props} style={[props.styles.input, props.style]} />
}

const getStylesFromProps = ({ theme }) => {
  return {
    input: {
      ...theme.fontStyle,
      fontFamily: theme.fonts.slab,
      color: theme.colors.darkGray,
      borderBottomStyle: 'solid',
      borderBottomWidth: StyleSheet.hairlineWidth,
      padding: theme.sizes.defaultHalf,
      borderBottomColor: theme.colors.darkGray,
    },
  }
}

export default withStyles(getStylesFromProps)(InputText)
