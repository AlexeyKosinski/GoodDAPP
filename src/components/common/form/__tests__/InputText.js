import React from 'react'
import renderer from 'react-test-renderer'
import InputText from '../InputText'
import { withThemeProvider } from '../../../../__tests__/__util__'

// Note: test renderer must be required after react-native.

describe('InputText', () => {
  const WrappedInputText = withThemeProvider(InputText)

  it('renders without errors', () => {
    const tree = renderer.create(<WrappedInputText />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('empty matches snapshot', () => {
    const component = renderer.create(<WrappedInputText />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches snapshot', () => {
    const component = renderer.create(<WrappedInputText value="Text" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
