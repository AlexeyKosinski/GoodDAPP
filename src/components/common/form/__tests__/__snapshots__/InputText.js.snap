// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText empty matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxFlex": 1,
      "WebkitFlexBasis": "0%",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
    }
  }
>
  <input
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect="on"
    className="css-textinput-1cwyjr8"
    data-focusable={true}
    dir="auto"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onSelect={[Function]}
    readOnly={false}
    spellCheck={true}
    style={
      Object {
        "borderBottomColor": "rgba(66,69,74,1.00)",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "1px",
        "color": "rgba(66,69,74,1.00)",
        "fontFamily": "RobotoSlab-Regular",
        "fontSize": "18px",
        "paddingBottom": "4px",
        "paddingLeft": "4px",
        "paddingRight": "4px",
        "paddingTop": "4px",
        "textAlign": "center",
      }
    }
    styles={
      Object {
        "input": Object {
          "borderBottomColor": "#42454A",
          "borderBottomStyle": "solid",
          "borderBottomWidth": 1,
          "color": "#42454A",
          "fontFamily": "RobotoSlab-Regular",
          "fontSize": 18,
          "padding": 4,
          "textAlign": "center",
        },
      }
    }
    theme={
      Object {
        "backdrop": Object {
          "backgroundColor": "rgba(255, 0, 0, 0.8)",
        },
        "borders": Object {
          "defaultBorderColor": "#c9c8c9",
        },
        "colors": Object {
          "accent": "#03dac4",
          "backdrop": "rgba(0, 0, 0, 0.5)",
          "background": "#f6f6f6",
          "blue": "#006EA0",
          "darkBlue": "#0C263D",
          "darkGray": "#42454A",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "error": "#FA6C77",
          "gray": "#555",
          "gray50Percent": "#CBCBCB",
          "gray80Percent": "#A3A3A3",
          "green": "#00C3AE",
          "lightGray": "#EEE",
          "lighterGray": "#A3A3A3",
          "notification": "#f50057",
          "orange": "#F8AF40",
          "placeholder": "#CBCBCB",
          "primary": "#00AFFF",
          "purple": "#9F6A9D",
          "red": "#FA6C77",
          "surface": "#ffffff",
          "text": "#222",
        },
        "dark": false,
        "feedItems": Object {
          "borderRadius": 8,
          "height": 84,
          "itemBackgroundColor": "#fff",
        },
        "fontStyle": Object {
          "color": "#555",
          "fontSize": 18,
          "textAlign": "center",
        },
        "fonts": Object {
          "bold": "Roboto-Bold",
          "light": "Roboto-Light, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "medium": "Roboto-Medium",
          "regular": "Roboto",
          "slab": "RobotoSlab-Regular",
          "slabBold": "RobotoSlab-Bold",
          "thin": "Roboto-Thin, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
        },
        "modals": Object {
          "backgroundColor": "#fff",
          "borderLeftWidth": 10,
          "borderRadius": 5,
          "contentPadding": 16,
          "jaggedEdgeSize": 15,
          "overlayBackgroundColor": "rgba(255, 255, 255, 0.8)",
          "overlayHorizontalPadding": 20,
          "overlayVerticalPadding": 30,
        },
        "paddings": Object {
          "defaultMargin": 8,
          "mainContainerPadding": 8,
        },
        "roundness": 4,
        "sizes": Object {
          "borderRadius": 5,
          "default": 8,
          "defaultDouble": 16,
          "defaultHalf": 4,
        },
      }
    }
    type="text"
  />
</div>
`;

exports[`InputText matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxFlex": 1,
      "WebkitFlexBasis": "0%",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
    }
  }
>
  <input
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect="on"
    className="css-textinput-1cwyjr8"
    data-focusable={true}
    dir="auto"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onSelect={[Function]}
    readOnly={false}
    spellCheck={true}
    style={
      Object {
        "borderBottomColor": "rgba(66,69,74,1.00)",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "1px",
        "color": "rgba(66,69,74,1.00)",
        "fontFamily": "RobotoSlab-Regular",
        "fontSize": "18px",
        "paddingBottom": "4px",
        "paddingLeft": "4px",
        "paddingRight": "4px",
        "paddingTop": "4px",
        "textAlign": "center",
      }
    }
    styles={
      Object {
        "input": Object {
          "borderBottomColor": "#42454A",
          "borderBottomStyle": "solid",
          "borderBottomWidth": 1,
          "color": "#42454A",
          "fontFamily": "RobotoSlab-Regular",
          "fontSize": 18,
          "padding": 4,
          "textAlign": "center",
        },
      }
    }
    theme={
      Object {
        "backdrop": Object {
          "backgroundColor": "rgba(255, 0, 0, 0.8)",
        },
        "borders": Object {
          "defaultBorderColor": "#c9c8c9",
        },
        "colors": Object {
          "accent": "#03dac4",
          "backdrop": "rgba(0, 0, 0, 0.5)",
          "background": "#f6f6f6",
          "blue": "#006EA0",
          "darkBlue": "#0C263D",
          "darkGray": "#42454A",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "error": "#FA6C77",
          "gray": "#555",
          "gray50Percent": "#CBCBCB",
          "gray80Percent": "#A3A3A3",
          "green": "#00C3AE",
          "lightGray": "#EEE",
          "lighterGray": "#A3A3A3",
          "notification": "#f50057",
          "orange": "#F8AF40",
          "placeholder": "#CBCBCB",
          "primary": "#00AFFF",
          "purple": "#9F6A9D",
          "red": "#FA6C77",
          "surface": "#ffffff",
          "text": "#222",
        },
        "dark": false,
        "feedItems": Object {
          "borderRadius": 8,
          "height": 84,
          "itemBackgroundColor": "#fff",
        },
        "fontStyle": Object {
          "color": "#555",
          "fontSize": 18,
          "textAlign": "center",
        },
        "fonts": Object {
          "bold": "Roboto-Bold",
          "light": "Roboto-Light, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "medium": "Roboto-Medium",
          "regular": "Roboto",
          "slab": "RobotoSlab-Regular",
          "slabBold": "RobotoSlab-Bold",
          "thin": "Roboto-Thin, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
        },
        "modals": Object {
          "backgroundColor": "#fff",
          "borderLeftWidth": 10,
          "borderRadius": 5,
          "contentPadding": 16,
          "jaggedEdgeSize": 15,
          "overlayBackgroundColor": "rgba(255, 255, 255, 0.8)",
          "overlayHorizontalPadding": 20,
          "overlayVerticalPadding": 30,
        },
        "paddings": Object {
          "defaultMargin": 8,
          "mainContainerPadding": 8,
        },
        "roundness": 4,
        "sizes": Object {
          "borderRadius": 5,
          "default": 8,
          "defaultDouble": 16,
          "defaultHalf": 4,
        },
      }
    }
    type="text"
    value="Text"
  />
</div>
`;
